src/lib/components/Switch/thumb.svelte

```diff
@@ -1,11 +1,13 @@
-<script lang="ts">
-	import { getState, getSwitchContext } from './root.svelte';
+<script lang="ts" context="module">
+	import type { BaseProps } from '$lib/types';
+	import { getSwitchContext, getState } from './root.svelte';
+
+	export type Props = BaseProps<HTMLSpanElement>;
```

Comment:
IMO if we're exporting, should be SwitchThumbProps, and we can also export it in index